src/python/pants/help/help_printer.py

```diff
@@ -91,10 +93,37 @@ def _print_goals_help(self) -> None:
                 {goal.name: goal.description_first_line for goal in Goal.all() if goal.description}
             )
 
-        max_width = max(len(name) for name in goal_descriptions.keys()) if goal_descriptions else 0
-        for name, description in sorted(goal_descriptions.items()):
-            print(f"  {name.rjust(max_width)}: {description}")
-        print()
+        title_text = "Goals"
+        title = f"{title_text}\n{'-' * len(title_text)}"
+        if self._use_color:
+            title = green(title)
+
+        max_width = max(len(name) for name in goal_descriptions.keys())
+        chars_before_description = max_width + 2
+
+        def format_goal(name: str, description: str) -> str:
+            name = name.ljust(chars_before_description)
+            if self._use_color:
+                name = cyan(name)
+            description_lines = textwrap.wrap(description, 80 - chars_before_description)
```

Comment:
I would prefer for this `80` to be demagicked

Reply:
One followup PR I'm planning is to add a util so that it's easy for us to consistently wrap exception messages, whereas we currently almost always use soft wrapping. I've changed my mind that I now think hard wrapping is a better UX in a CLI app than soft wrapping.

I think we'd want the 80 demagicking to happen in that PR. Right now, we're duplicating `80` in 2 places, and once I add `./pants backends`, 3 places. Imo, it's premature to factor out the `80` constant without also factoring out the overall wrapping mechanism in a followup.